back/src/domains/establishment/adapters/PgEstablishmentAggregateRepository.ts

```diff
@@ -667,31 +660,34 @@ export class PgEstablishmentAggregateRepository
     inseeCheckDate: Date,
     params: UpdateEstablishmentsWithInseeDataParams,
   ): Promise<void> {
-    const queries = keys(params).map((siret) => {
-      const values = params[siret];
-      return format(
-        `
-            UPDATE establishments
-            SET last_insee_check_date = %1$L 
-              ${values?.isOpen !== undefined ? ", is_open=%3$L" : ""}
-              ${values?.nafDto ? ", naf_code=%4$L" : ""}
-              ${values?.nafDto ? ", naf_nomenclature=%5$L" : ""}
-              ${values?.name ? ", name=%6$L" : ""}
-              ${values?.numberEmployeesRange ? ", number_employees=%7$L" : ""}
-            WHERE siret = %2$L;`,
-        inseeCheckDate.toISOString(),
-        siret,
-        ...[
-          values?.isOpen,
-          values?.nafDto?.code,
-          values?.nafDto?.nomenclature,
-          values?.name,
-          values?.numberEmployeesRange,
-        ],
-      );
+    toPairs(params).forEach(async ([siret, values]) => {
```

Comment:
`.forEach` ça fonctionne très mal avec les promises. Je te recommande de regarder `executeInSequence` (dans `utils.ts`) qui se base sur `for ()`.